multicast/hear.py

```diff
@@ -0,0 +1,368 @@
+		B: Test that the hear component is initialized.
+		C: Test that the main(HEAR) function-flow is initialized.
+		D: Test that the main(HEAR) function-flow returns an int 0-3.
+
+		>>> multicast.__main__ is not None
+		True
+		>>> multicast.__main__.main is not None
+		True
+		>>> tst_fxtr_args = ['''HEAR''', '''--port=1234''']
+		>>> (test_fixture, ignored_value) = multicast.__main__.main(tst_fxtr_args)
+		>>> test_fixture is not None
+		True
+		>>> type(test_fixture) #doctest: -DONT_ACCEPT_BLANKLINE, +ELLIPSIS
+		<...int...>
+		>>> int(test_fixture) >= int(0)
+		True
+		>>> type(test_fixture) is type(0)
+		True
+		>>> int(test_fixture) < int(4)
+		True
+		>>> (int(test_fixture) >= int(0)) and (int(test_fixture) < int(4))
+		True
+		>>>
+
+"""
+
+
+__package__ = """multicast"""
+"""Names the package of this program.
+
+	Minimal Acceptance Testing:
+
+	First setup test fixtures by importing multicast.
+
+	Testcase 0: Multicast should be importable.
+
+		>>> import multicast
+		>>>
+
+	Testcase 1: Hear should be automaticly imported.
+
+		>>> multicast.hear.__package__ is not None
+		True
+		>>>
+		>>> multicast.hear.__package__ == multicast.__package__
+		True
+		>>>
+
+"""
+
+
+__module__ = """multicast"""
+"""Names the module of this program.
+
+	Minimal Acceptance Testing:
+
+	First setup test fixtures by importing multicast.
+
+	Testcase 0: Multicast should be importable.
+
+		>>> import multicast
+		>>>
+
+	Testcase 1: Hear should be automaticly imported.
+
+		>>> multicast.hear.__module__ is not None
+		True
+		>>>
+
+"""
+
+
+__file__ = """multicast/hear.py"""
+"""Names the file of this component."""
+
+
+__name__ = """multicast.hear"""
+"""Names this component.
+
+	Minimal Acceptance Testing:
+
+	First setup test fixtures by importing multicast.
+
+	Testcase 0: Multicast should be importable.
+
+		>>> import multicast
+		>>>
+
+	Testcase 1: Hear should be automaticly imported.
+
+		>>> multicast.hear.__name__ is not None
+		True
+		>>>
+
+"""
+
+try:
+	import sys
+	if 'multicast' not in sys.modules:
+		from . import multicast as multicast
+	else:  # pragma: no branch
+		multicast = sys.modules["""multicast"""]
+	_BLANK = multicast._BLANK
+	from . import recv as recv
+	from . import send as send
+except Exception as importErr:
+	del importErr
+	import multicast as multicast
+
+
+try:
+	import socketserver
+	from socketserver import threading as threading
+	from multicast import argparse as argparse
+	from multicast import unicodedata as unicodedata
+	from multicast import socket as socket
+	from multicast import struct as struct
+	depends = [
+		unicodedata, socket, struct, argparse
+	]
+	for unit in depends:
+		try:
+			if unit.__name__ is None:  # pragma: no branch
+				raise ImportError(
+					str("[CWE-440] module failed to import {}.").format(str(unit))
+				)
+		except Exception:  # pragma: no branch
+			raise ImportError(str("[CWE-758] Module failed completely."))
+except Exception as err:
+	raise ImportError(err)
```

Comment:
This suggestion would cause a regression in some python environments, and does not account for the historical changes to differing python versions and the uncertainty of how many versions of multicast are currently installed (a very real issue in development). 
It may be helpful to understand that the first try-except is considered handled by the:
> except Exception as importErr:
>    del importErr
>    import multicast as multicast

however the second and its nested try-excepts are NOT considered handled and must pass the responsibility of handling this error up to the caller (of the whole python file when importing) for example a case or out-of-order imports could be resolved. This is less efficient in the more common/modern python environments but provides a more robust range of portability.

Any suggestions to the import structure must at a minimum support the same try-except scope of the current implementation.

In summation the preference is for portability when in conflict with clarity.

Caveat: Code is often imperfect though.

Please try again with more consideration for maintaining cross-python compatibly.